Base/src/main/java/io/deephaven/base/FileUtils.java

```diff
@@ -59,19 +61,23 @@ public static void cleanDirectory(File path) {
     }
 
     public static void deleteRecursively(File file) {
-        if (!file.exists()) {
+        if (!Files.exists(file.toPath(), LinkOption.NOFOLLOW_LINKS)) {
             return;
         }
         if (file.isDirectory()) {
-            File f[] = file.listFiles();
+            final File[] files = file.listFiles();
```

Comment:
I wonder if we might prefer

```java
        Files.walkFileTree(file.toPath(), new SimpleFileVisitor<>() {
            @Override
            public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
                if (exc != null) {
                    throw exc;
                }
                Files.deleteIfExists(dir);
                return FileVisitResult.CONTINUE;
            }

            @Override
            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
                Files.deleteIfExists(file);
                return FileVisitResult.CONTINUE;
            }
        });
```

it looks like walkFileTree does not follow links by default.

Reply:
That isn't clearly better and would require testing it vs. the ported code which has already been tested.

Reply:
The only thing that makes it better if it's not recursive.